### ISSUE TITLE
layers.Inputlayer does not show up in model.summary()

### ISSUE THREAD

**Issue by rohanreddych** [tensorflow/tensorflow#35515]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
Tensorflow 2.1.0 on gpu on colab

**Describe the current behavior**
In model.summary(), layers.Inputlayer() does not show up as a layer.

**Code to reproduce the issue**
`model1 = tf.keras.models.Sequential([layers.InputLayer(input_shape=(784,)),
                                     layers.Dense(1500, activation="relu"),
                                     layers.Dense(1000, activation="relu"),
                                     layers.Dense(500, activation="relu"),
                                     layers.Dense(1, activation="sigmoid"),
                                     layers.Dense(500, activation="relu"),
                                     layers.Dense(1000, activation="relu"),
                                     layers.Dense(1500, activation="relu"),
                                     layers.Dense(784),
])`
![Screenshot from 2019-12-31 22-39-50_modified](https://user-images.githubusercontent.com/43641071/71628625-d147b500-2c1e-11ea-873a-b8adf4259cfd.png)

**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by Lasers67** [comment]
the Input layer does not show up explicitly in model.summary() when the Sequential API is used does not mean anything regarding the functionality of the model.
refer:-
[https://stackoverflow.com/questions/54497422/the-input-layer-disappears-from-the-structure-of-a-deep-learning-model](url)

**Comment by rohanreddych** [comment]
i know it doesn't affect the functionality but it does show up in the keras functional api so i thought it would be a bug in tensorflow

**Comment by ravikyram** [comment]
@rohanreddych 

I have tried in colab with TF 2.1 using TF.keras and i am not seeing ` layers.Inputlayer() `in` model1.summary()`. Please, find the gist [here.](https://colab.sandbox.google.com/gist/ravikyram/3579f1602b4bceb0d7a526cee3316e86/untitled521.ipynb).I have tried with TF 1.14 in Keras as well ,still i am not seeing  `layers.Inputlayer()` in `model1.summary()`. Please, find the gist [here](https://colab.sandbox.google.com/gist/ravikyram/6971f8ce63cae4b4a29904721c4fad43/untitled522.ipynb).Thanks!

**Comment by rohanreddych** [comment]
Yes it is consistent with my issue so is it a bug in tensorflow

**Comment by ymodak** [comment]
Input layer merely decides the input shape and is not necessarily a layer. 
You may try using ```functional api``` to explicitly see it in model summary:,
```python
from tensorflow.keras.models import Model
from tensorflow.keras.layers import Input, Dense  # explicitly import Input layer

inputs = Input(shape=(784,))
x = Dense(1500, activation="relu")(inputs)
x = Dense(1000, activation="relu")(x)
prediction = Dense(784, activation='softmax')(x)

model_functional = Model(inputs=inputs, outputs=prediction)

model_functional.summary()
```
Output:
```python
Model: "model"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
input_1 (InputLayer)         [(None, 784)]             0         
_________________________________________________________________
dense (Dense)                (None, 1500)              1177500   
_________________________________________________________________
dense_1 (Dense)              (None, 1000)              1501000   
_________________________________________________________________
dense_2 (Dense)              (None, 784)               784784    
=================================================================
Total params: 3,463,284
Trainable params: 3,463,284
Non-trainable params: 0
_________________________________________________________________
```

**Comment by noppelmax** [comment]
Suppose I want to implement backprop on my own (independent if the model is created with functional or sequential API), how do I know if the input layer is included or not?! Why can't it just be implemented consistent with the functional API? Argh

In `model.get_config()["layers"]` the input layer is still listed, while in `model.layers` it is not present?! In a functional model it is present in both.

```
import tensorflow as tf

seqmodel = tf.keras.Sequential()
seqmodel.add(tf.keras.Input(shape=(4)))
seqmodel.add(tf.keras.layers.Dense(1,activation="sigmoid"))

inLayer = tf.keras.layers.Input(shape=(4))
denseLayer = tf.keras.layers.Dense(1,activation="sigmoid")
out = denseLayer(inLayer)
funcmodel = tf.keras.Model(inputs=inLayer, outputs=out, name="twolayer")

print("Seq Config Num Layers: ",len(seqmodel.get_config()["layers"]))
print("Func Config Num Layers: ",len(funcmodel.get_config()["layers"]))
print()
print("Seq Num Layers : ",len(seqmodel.layers))
print("Func Num Layers: ",len(funcmodel.layers))
```

Output:
```
Seq Config Num Layers:  2
Func Config Num Layers:  2

Seq Num Layers :  1
Func Num Layers:  2
```

This leads to this fix:
```
    layers = model.layers
    numLayers = len(layers)
    
    if len(model.get_config()["layers"]) != len(model.layers):
        sequential = True
        offset = 0
        print("Sequential Model")
    else:
        sequential = False
        offset = 1
        print("Functional Model")
        
    for nlid in range(0,numLayers-offset): ....
```

**Comment by japm48** [comment]
I hope that the fact that this is not consistent with the functional API motivates to implement this.

Also related: issue #35995.

**Comment by RafalSkolasinski** [comment]
> Input layer merely decides the input shape and is not necessarily a layer.
> You may try using `functional api` to explicitly see it in model summary:,
> 
> ```python
> from tensorflow.keras.models import Model
> from tensorflow.keras.layers import Input, Dense
> 
> inputs = Input(shape=(784,))
> x = Dense(1500, activation="relu")(inputs)
> x = Dense(1000, activation="relu")(x)
> prediction = Dense(784, activation='softmax')(x)
> 
> model_functional = Model(inputs=inputs, outputs=prediction)
> 
> model_functional.summary()
> ```
> 
> Output:
> 
> ```python
> Model: "model"
> _________________________________________________________________
> Layer (type)                 Output Shape              Param #   
> =================================================================
> input_1 (InputLayer)         [(None, 784)]             0         
> _________________________________________________________________
> dense (Dense)                (None, 1500)              1177500   
> _________________________________________________________________
> dense_1 (Dense)              (None, 1000)              1501000   
> _________________________________________________________________
> dense_2 (Dense)              (None, 784)               784784    
> =================================================================
> Total params: 3,463,284
> Trainable params: 3,463,284
> Non-trainable params: 0
> _________________________________________________________________
> ```

Why this kind of summary is not a default one? It would be so much more informative.

**Comment by SaiRishab24** [comment]
The reason for this is we will run the model.summary() command before we fit and run the model where input will be mentioned. So basically the model.summary wont be having any idea of input dimension at its time. so in sequential you cant see the input layer in summary . hope this helps

**Comment by helderdaniel** [comment]
@SaiRishab24 this code line defines the input shape, thus model.summary should have an idea:

inputs = Input(shape=(784,))
